CorsixTH/Lua/dialogs/staff_dialog.lua

```diff
@@ -314,12 +314,7 @@ end
 
 function UIStaff:placeStaff()
   self.staff.pickup = true
-  self.staff:setNextAction({
-    name = "pickup",
-    ui = self.ui,
-    todo_close = self,
-    must_happen = true,
-  }, true)
+  self.staff:setNextAction(PickupAction():setUi(self.ui):setTodoClose(self):setMustHappen(), true)
```

Comment:
What do you think about mixing the highPriority flag into the action, i.e. :setHighPriority() instead of ,true?

(out of scope but the queueAction position number bugs me a bit too, seems like leaky abstraction)